handbook/engineering/hiring/software-engineer-code-intelligence-indexing.md

```diff
@@ -0,0 +1,84 @@
+## Software Engineer - Code Intelligence Indexing
+
+The Code Intelligence [team](../code-intelligence/index.md) builds tools and services that provide contextual information around code, taking into account its lexical, syntactic, and semantic structure.  We want to provide users with fast, comprehensive and accurate answers to important code navigation queries.  
+
+Our mission is to build a code intelligence platform that understands code in all languages and provides rich metadata to power [code search](https://docs.sourcegraph.com/code_search) and  [campaigns](https://docs.sourcegraph.com/campaigns) alongside future features.
+
+
+
+## How does Code Intelligence work from a high level?
+
+We write language specific indexers that emit [LSIF](https://lsif.dev/) data about projects.  This information is processed, persisted and used to answer [LSP](https://microsoft.github.io/language-server-protocol/) requests on demand such as Go to Definition and Find References within Sourcegraph.
+
+## What problems would you be working on?
+
+Our current [focus](../code-intelligence/goals.md) is to provide automated code intelligence for C++ and Java to supplement our Go and TS indexers.  
+
+We're looking for someone to lead either the C++ or Java development of the LSIF indexers. Problems that we want to solve include:
+
+* Automatic support for a variety of dependency management and builds systems in each ecosystem.
+* Incremental indexing, currently a one line change will need a full re-index of a project.  Help us make this faster and more efficient.
+* Resolution and indexing of 3rd party dependencies
+* Distributed builds
+
+
+
+As our mission is to provide Code Intelligence for all languages, this role will also be crucial in setting a foundation for how we approach indexing across a broad variety of languages.
+
+
+
+## What will the team potentially look like in a year from now?
+
+We will grow to 6 to 8 engineers, with two distinctive problem spaces:
+
+* 3-4 engineers focusing on the indexers, expanding support for a wider range of languages (think the rest of the C/JVM family and Python etc).
+* 3-4 engineers transforming and exposing the LSIF data, optimizing queries to be as fast as possible.
+
+
+
+## Qualifications
+
+* Solid understanding of either C++ or Java with a focus on Clang or Javac fundamentals from a language tooling perspective.
+* Experience with dependency management systems and build systems coupled with a strong desire to learn new ones.
+* Ability to communicate clearly and empathetically, especially in writing and documentation.
+* Practiced at creating high quality software balanced with a pragmatic understanding of how to make appropriate tradeoffs (e.g., cut scope) to ship quickly and iterate when necessary.
```

Comment:
These aren't unique to the role and don't really fit the definition of a qualification: https://about.sourcegraph.com/handbook/people-ops/hiring/job_description_guidelines#qualifications


```suggestion
```

See https://about.sourcegraph.com/handbook/engineering/hiring/engineering-manager-cloud for how I talked about our values in the "responsibilities" section, but this isn't a "qualification".